app/controllers/coronavirus_form/support_address_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::SupportAddressController < ApplicationController
+  include ActionView::Helpers::SanitizeHelper
+  include FieldValidationHelper
+
+  REQUIRED_FIELDS = %w(
+      building_and_street_line_1
+      town_city
+  ).freeze
+
+  def show
+    session[:support_address] ||= {}
+    render "coronavirus_form/#{PAGE}"
+  end
+
+  def submit
+    session[:support_address] ||= {}
+    session[:support_address]["building_and_street_line_1"] = sanitize(params[:building_and_street_line_1]).presence
+    session[:support_address]["building_and_street_line_2"] = sanitize(params[:building_and_street_line_2]).presence
+    session[:support_address]["town_city"] = sanitize(params[:town_city]).presence
+    session[:support_address]["county"] = sanitize(params[:county]).presence
+    session[:support_address]["postcode"] = sanitize(params[:postcode]).presence
+
+    invalid_fields = validate_fields(session[:support_address])
+
+    if invalid_fields.any?
+      flash.now[:validation] = invalid_fields
+      render "coronavirus_form/#{PAGE}"
+    elsif session["check_answers_seen"]
+      redirect_to controller: "coronavirus_form/check_answers", action: "show"
+    else
+      redirect_to controller: "coronavirus_form/#{NEXT_PAGE}", action: "show"
+    end
+  end
+
+private
+
+  PAGE = "support_address"
+  NEXT_PAGE = "contact_details"
+
+  def validate_fields(support_address)
+    [
+      validate_missing_fields(support_address),
+      validate_conditionally_present_fields(support_address),
+    ].flatten.uniq
+  end
+
+  def validate_conditionally_present_fields(product)
+    return [] if product["postcode"].present? || product["county"].present?
+
+    invalid_fields = []
+
+    if product["postcode"].blank?
+      invalid_fields << {
+        field: "postcode",
+        text: t("coronavirus_form.errors.missing_postcode_or_county_fields"),
```

Comment:
Given we have both postcode and county fields, this two-in-one error message feels confusing. 🤔

Reply:
ah i see, like keep the message but split it out?
I'm still waiting on final copy - but the deal is you must have at least one of them

Reply:
i've amended that to split them out, hopefully a bit less confusing?